### PR TITLE
Move applyConstraints algorithm off main thread, and handle errors.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fix for https://github.com/w3c/mediacapture-main/issues/530.

### CODE REVIEW

**Comment by jan-ivar** [comment]
[Diff](https://github.com/w3c/mediacapture-main/pull/531/files?w=1) with whitespace ignored.

File: getusermedia.html [hunk 1]
```diff
@@ -4274,48 +4316,28 @@ <h2>Methods</h2>
                   set to <code>"none"</code> over <code>"crop-and-scale"</code>.</p>
                 </li>
               </ol>
-              <p>When the <dfn>applyConstraints algorithm</dfn> is called, the
-              User Agent MUST run the following steps:</p>
+              <p>To apply the <dfn>applyConstraints algorithm</dfn> to an
```
Comment (henbos) [inline getusermedia.html]: nit: applyConstraints vs ApplyConstraints

File: getusermedia.html [hunk 2]
```diff
@@ -4127,8 +4130,47 @@ <h2>Methods</h2>
...
+                </li>
+                <li>
+                  <p>Let <var>p</var> be a new promise.</p>
+                </li>
+                <li>
+                  <p>Run the following steps in parallel:</p>
+                  <ol>
+                    <li>
+                      <p>Let <var>failedConstraint</var> be the result of
+                      running the <a>applyConstraints algorithm</a> with
```
Comment (henbos) [inline getusermedia.html]: nit: This is referred to as <a>ApplyConstraints algorithm</a> above, which is consistent with the capitalized "SelectSettings algorithm".

File: getusermedia.html [hunk 3]
```diff
@@ -4127,8 +4130,47 @@ <h2>Methods</h2>
...
+                      <var>message</var>)</code>, and abort these steps. The
+                      existing constraints remain in effect in this case.</p>
+                    </li>
+                  </ol>
+                </li>
+                <li>
+                  <p>Return <var>p</var>.</p>
+                </li>
+              </ol>
               <p>The <a>applyConstraints algorithm</a> for applying constraints
```
Comment (henbos) [inline getusermedia.html]: nit: applyConstraints vs ApplyConstraints

**Pull request merged**